### PR TITLE
Document the rest of the owl

### LINKED ISSUE #1076: Document Config
- [ ] Document the public fields
- [ ] Document the impl

### LINKED ISSUE #1078: Document the EncoderConfig

### LINKED ISSUE #1349: Copy back documentation from crav1e api to native rust API
The crav1e documentation is more complete at the moment.

### LINKED ISSUE #1484: Fully document API modules
Depends on:
- [x] #1075
- [ ] #1076
- [ ] #1078
- [x] #1079
And more.

This is considered resolved when both `api.rs` and `capi.rs` have the `#![deny(missing_docs)]` attribute.

### PR DESCRIPTION
**Pull request by YaLTeR**
Closes #1076, closes #1078, closes #1484, closes #1349, closes #1219.

Adds `#![deny(missing_docs)]` to capi and `src/api/mod.rs`.

### CODE REVIEW

File: src/api/mod.rs [hunk 1]
```diff
@@ -42,81 +43,117 @@ use std::sync::Arc;
...
+  ///
+  /// [`tile_cols`]: #structfield.tile_cols
+  /// [`tile_rows`]: #structfield.tile_rows
   pub tiles: usize,
-  /// Number of frames to read ahead for RDO lookahead computation.
+  /// Number of frames to read ahead for the RDO lookahead computation.
   pub rdo_lookahead_frames: u64,
+  /// Settings which affect the enconding speed vs. quality trade-off.
   pub speed_settings: SpeedSettings,
+  /// Enables printing the PSNR values.
```
Comment (dwbuiten) [inline src/api/mod.rs]: In terms of the API, it doesn't pint anything - it just attaches the PSNR to the output packet.

File: src/api/mod.rs [hunk 2]
```diff
@@ -380,6 +450,21 @@ fn check_tile_log2(n: usize) -> bool {
...
+  /// Creates a [`Context`] with this configuration.
+  ///
+  /// # Examples
+  ///
+  /// ```
+  /// use rav1e::prelude::*;
+  ///
+  /// # fn main() -> Result<(), EncoderStatus> {
+  /// let cfg = Config::default();
+  /// let ctx: Context<u8> = cfg.new_context()?;
```
Comment (dwbuiten) [inline src/api/mod.rs]: .unwrap may be needed (?)

File: src/capi.rs [hunk 3]
```diff
@@ -537,6 +545,7 @@ pub unsafe extern fn rav1e_frame_new(ctx: *const Context) -> *mut Frame {
   Box::into_raw(frame)
 }
 
+/// Free the Frame.
```
Comment (dwbuiten) [inline src/capi.rs]: I would change all the unref functions to something like `/// Free a RaFrame`.

File: src/capi.rs [hunk 4]
```diff
@@ -244,6 +252,7 @@ pub unsafe extern fn rav1e_data_unref(data: *mut Data) {
   }
 }
 
+/// Create a Config filled with default parameters.
```
Comment (dwbuiten) [inline src/capi.rs]: RaConfig

**Comment by lu-zero** [comment]
The crate got updated with your improvement :) Feel free to rebase + bump the dependency

File: src/util.rs [hunk 5]
```diff
@@ -123,11 +125,15 @@ impl_cast_from_primitive!(i16 => { i8, i16, i32, i64, isize });
 impl_cast_from_primitive!(i32 => { u32, u64, usize });
 impl_cast_from_primitive!(i32 => { i8, i16, i32, i64, isize });
 
+/// Types that can be used as pixel types.
 pub enum PixelType {
+  /// `u8`
```
Comment (lu-zero) [inline src/util.rs]:
```suggestion
  /// 8bits pixel, stored as `u8`.
```

File: src/util.rs [hunk 6]
```diff
@@ -123,11 +125,15 @@ impl_cast_from_primitive!(i16 => { i8, i16, i32, i64, isize });
...
   U8,
+  /// `u16`
```
Comment (lu-zero) [inline src/util.rs]:
```suggestion
  /// 10 or 12bits per pixel, stored as `u16`.
```

**Pull request merged**